src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/mapper/YFlowMapper.java

```diff
@@ -110,4 +108,31 @@ protected Collection<Flow> getDiverseWithFlow(Flow flow, FlowRepository flowRepo
                                 || !flow.getYFlowId().equals(diverseFlow.getYFlowId()))
                         .collect(Collectors.toSet());
     }
+
+    private Set<String> getDiverseHaFlowIds(String diversityTyGroup, HaFlowRepository haFlowRepository) {
+        if (diversityTyGroup == null) {
+            return Collections.emptySet();
+        }
+        return new HashSet<>(haFlowRepository.findHaFlowsIdByDiverseGroupId(diversityTyGroup));
+    }
+
+    /**
+     * Finds diverse flow ids in the list of diverse flows.
+     */
+    public static Set<String> getDiverseFlowIds(Collection<Flow> diverseFlows) {
+        return diverseFlows.stream()
+                .filter(f -> f.getYFlowId() == null)
+                .map(Flow::getFlowId)
+                .collect(Collectors.toSet());
+    }
+
+    /**
+     * Finds diverse y-flow ids in the list of diverse flows.
+     */
+    public static Set<String> getDiverseYFlowIds(Collection<Flow> diverseWithFlow) {
+        return diverseWithFlow.stream()
+                .map(Flow::getYFlowId)
+                .filter(Objects::nonNull)
```

Comment:
on 124 you don't do unnecessary mapping, but 135 looks nicer 🤔 I vote for 124, though. 

_(If only java knew the map operation is stateless, it would optimize it to the same bytecode)_